File: OlderCodes/Fortran77codes/sqwell.f
c  From: "A SURVEY OF COMPUTATIONAL PHYSICS" 
c  by RH Landau, MJ Paez, and CC BORDEIANU 
c  Copyright Princeton University Press, Princeton, 2008.
c  Electronic Materials copyright: R Landau, Oregon State Univ, 2008;
c  MJ Paez, Univ Antioquia, 2008; and CC BORDEIANU, Univ Bucharest, 2008.
c  Support by National Science Foundation                             
c    			                        
c  sqwell.f: Solves the time-dependent Schroedinger equation for a 
c	           Gaussian wavepacket in a infinite square well potential
c			       Output data in gnuplot 3D format                    

      Program sqwell
			
        Implicit None
      	Real*8 psr(751, 2), psi(751, 2), p2(751)
       	Real*8 dx, k0, dt, x, pi
      	Integer i, n, max
      	Complex exc, zi
      	Common /values/dx, dt
      	Open(9, file = 'sqwell.dat', status = 'Unknown')
      	max = 750
      	pi = 3.14159265358979323846
      	zi = CMPLX(0.d0, 1.d0)
      	dx = 0.02d0
      	k0 = 17.d0*pi
      	dt = dx*dx   
c                                                  initial conditions
      	x = 0.
      	Do 10 i = 1, max+1
          exc = exp(zi*k0*x)
c                                    real, imag initial wave function
          psr(i, 1) = real(exc*exp(-0.5*(2.*(x-5.))**2.))
          psi(i, 1) = imag(exc*exp(-0.5*(2.*(x-5.))**2.))
          x = x + dx
 10   	Continue
c                                     propagate solution through time
      	Do 40 n = 1, 6000
c                                            real psi and probability
          Do 50 i = 2, max
            psr(i, 2) = psr(i, 1) - dt*(psi(i+1, 1) + psi(i-1, 1)
     1                 -2.*psi(i, 1))/(2.*dx*dx)
            p2(i) = psr(i, 1)*psr(i, 2)+psi(i, 1)*psi(i, 1)    
 50       Continue
c                                 imaginary part of the wave function
          Do 60 i = 2, max
            psi(i, 2) = psi(i, 1) + dt*(psr(i+1, 2) + psr(i-1, 2)
     1                -2.d0*psr(i, 2))/(2.d0*dx*dx)  
 60       Continue  
c                                            output  at certain times
    	  If ( Mod(n, 300) .eq. 0) Then 
          Do 80 i = 1, max+1, 15
            write(9, 11)p2(i)
 80       Continue   
          write(9, 11)   
          EndIf
c                                                          new -> old
          Do 70 i = 1, max+1
            psi(i, 1) = psi(i, 2)
            psr(i, 1) = psr(i, 2)
 70       Continue                     
 40   	Continue
 11	    Format (E12.6)
 	      Close(9)     
      	Stop 'data saved in sqwell.dat'
       End

File: OlderCodes/JavaCodes/sqwell.java
/* From: "A SURVEY OF COMPUTATIONAL PHYSICS" 
   by RH Landau, MJ Paez, and CC Bordeianu 
   Copyright Princeton University Press, Princeton, 2007.
   Electronic Materials copyright: R Landau, Oregon State Univ, 2007;
   MJ Paez, Univ Antioquia, 2007; & CC Bordeianu, Univ Bucharest, 2007
   Support by National Science Foundation                              
* 
*  sqwell.java: Solves the time-dependent Schroedinger equation for a 
* Gaussian wavepacket in a infinite square well potential            
*  comment: Output data is saved in 3D grid format used by gnuplot. 
*/
import java.io.*;

public class sqwell  {

public static void main(String[] argv) throws IOException,
FileNotFoundException
{
  PrintWriter w = 
        new PrintWriter(new FileOutputStream("sqwell.dat"), true);
double psr[][]=new double[751][2];
double psi[][]=new double[751][2];
double p2[]=new double[751];
double dx,k0,dt,x,pi;
int i,n,max;
        max   = 750;
        pi   = 3.14159265358979323846;
        dx   = 0.02;
        k0   = 17.0*pi;
        dt   = dx*dx;   
//initial conditions
        x   = 0.0;
        for( i=0;i<max;i++){
/* 
 compute real part and imaginary part of the initial wave
 function using Euler's formula: 
 exp(i*k0*x)=cos(k0*x)+i*sin(k0*x)        
*/
//real part of the initial wave function at t=0         
psr[i][0] = Math.exp(-0.5*(2.0*Math.pow(x-5.0,2.0)))*Math.cos(k0*x);
//imaginary part of the initial wave function at t=0            
psi[i][0] = Math.exp(-0.5*(2.0*Math.pow(x-5.0,2.0)))*Math.sin(k0*x);
          x        = x + dx;
                              }

//now propagate solution through time

        

//the real part of the wave function and the probability        
          
for(n=0;n<6000;n++){
  for(i=1;i<max-1;i++){
psr[i][1] = psr[i][0] - dt*(psi[i+1][0] + psi[i-1][0]-2.0*psi[i][0])/(2.0*dx*dx);
p2[i]    = psr[i][0]*psr[i][1]+psi[i][0]*psi[i][0];             
                        }
//the imaginary part of the wave function    
         
        for(i=1;i<max-1;i++){
psi[i][1] = psi[i][0] + dt*(psr[i+1][1] + psr[i-1][1]-2.0*psr[i][1])/(2.0*dx*dx);  
                            } 
//only at certain time instants we want to know
//the probability
if(n%300==0) { 
            
  for(i=0;i<max;i=i+15){             
              
    w.println(""+p2[i]+"");
                       }
           w.println("");   
              }
//new iterations are now the old ones
          
 for(i=0;i<max;i++){           
            psi[i][0] = psi[i][1];
            psr[i][0] = psr[i][1];
                   }                   
}
 
      
        System.out.println("data saved in sqwell.dat");
        

}
} 